tests/tagged_vector.cpp

```diff
@@ -61,3 +61,15 @@ TEST(TaggedVector, Transpose)
     ASSERT_EQ(coord.get<double>(), coord_reordered.get<double>());
     ASSERT_EQ(coord.get<float>(), coord_reordered.get<float>());
 }
+
+TEST(TaggedVector, Operators)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, float, double> b(3, 4);
+    ddc::detail::TaggedVector<int, double> c = ddc::select<double>(a);
+    ASSERT_EQ(a + b, (ddc::detail::TaggedVector<int, double, float>(5, 5)));
+    ASSERT_EQ(b - a, (ddc::detail::TaggedVector<int, double, float>(3, 1)));
+    ASSERT_EQ(c + 4, (ddc::detail::TaggedVector<int, double>(5)));
+    ASSERT_EQ(4 + c, (ddc::detail::TaggedVector<int, double>(5)));
+    ASSERT_EQ(4 * a, (ddc::detail::TaggedVector<int, double, float>(4, 8)));
```

Comment:
Many `ASSERT_*` are used in the tests. We should switch to their `EXPECT_*` equivalent.